src/test/java/com/artipie/test/TestDeployment.java

```diff
@@ -0,0 +1,215 @@
+ * Artipie container can be accessed from client container by {@code artipie} hostname.
+ * To run a command in client container and match a result use {@code assertExec} method.
+ * @since 0.19
+ */
+public final class TestDeployment implements BeforeEachCallback, AfterEachCallback {
+
+    /**
+     * Artipie builder.
+     */
+    private final Supplier<ArtipieContainer> asup;
+
+    /**
+     * Client builder.
+     */
+    private final Supplier<ClientContainer> csup;
+
+    /**
+     * Container network.
+     */
+    private Network net;
+
+    /**
+     * Artipie container.
+     */
+    private ArtipieContainer artipie;
+
+    /**
+     * Client container.
+     */
+    private ClientContainer client;
+
+    /**
+     * New container test.
+     * @param artipie Artipie container definition
+     * @param client Client container definition
+     */
+    public TestDeployment(final Supplier<ArtipieContainer> artipie,
+        final Supplier<ClientContainer> client) {
+        this.asup = artipie;
+        this.csup = client;
+    }
+
+    @Override
+    public void beforeEach(final ExtensionContext context) throws Exception {
+        this.net = Network.newNetwork();
+        this.artipie = this.asup.get()
+            .withNetwork(this.net).withNetworkAliases("artipie");
+        this.artipie = this.artipie.withLogConsumer(
+            new Slf4jLogConsumer(LoggerFactory.getLogger(this.artipie.getDockerImageName()))
+        );
+        this.client = this.csup.get()
+            .withNetwork(this.net)
+            .withCommand("tail", "-f", "/dev/null");
+        this.artipie.start();
+        this.client.start();
+    }
+
+    @Override
+    public void afterEach(final ExtensionContext context) throws Exception {
+        if (this.artipie != null) {
+            this.artipie.stop();
+            this.artipie.close();
+        }
+        if (this.client != null) {
+            this.client.stop();
+            this.client.close();
+        }
+        if (this.net != null) {
+            this.net.close();
+        }
+    }
+
+    /**
+     * Exec command in client container and assert result.
+     * @param msg Assertion message on failure
+     * @param matcher Exec result matcher
+     * @param cmd Command to execute
+     * @throws IOException In case of client exception
+     */
+    public void assertExec(final String msg, final Matcher<ExecResult> matcher,
+        final String... cmd) throws IOException {
+        final ExecResult exec;
+        try {
+            exec = this.client.execInContainer(cmd);
+        } catch (final InterruptedException ignore) {
+            Thread.currentThread().interrupt();
+            return;
+        }
+        MatcherAssert.assertThat(msg, exec, matcher);
+    }
+
+    /**
+     * Put resource to artipie server.
+     * @param res Resource path
+     * @param path Artipie path
+     */
+    public void putResourceArtipie(final String res, final String path) {
```

Comment:
@g4s8 this method name sounds odd, I'd suggest either add `To` or put `artipie` in the middle: `putArtipieResource`